### PR TITLE
AIMS-258, AIMS-259 Add activity loggers for api scan/send item

### PR DESCRIPTION
**Pull request by rfox2**
**What**: Added loggers for API scan/send ops

**How**: Added loggers to the activity logger class, added tests, and added log_activity method to api_post_deliver_item service class

### CODE REVIEW

File: app/services/api_post_deliver_item.rb [hunk 1]
```diff
@@ -21,6 +22,15 @@ def post_data!
 
   private
 
+  def log_activity(delivery_type, params, response)
```
Comment (jaronkk) [inline app/services/api_post_deliver_item.rb]: delivery_type and params are both available through private methods by the same name, so you don't need to pass them as arguments here

File: app/services/api_post_deliver_item.rb [hunk 2]
```diff
@@ -21,6 +22,15 @@ def post_data!
...
+    case delivery_type
+    when "scan"
+      ActivityLogger.api_scan_item(item: match.item, params: params, api_response: response)
```
Comment (jaronkk) [inline app/services/api_post_deliver_item.rb]: there is an item private method that you can make use of, instead of having to access match.item

**Comment by jgondron** [comment]
:+1:

**Comment by jaronkk** [comment]
A couple things: 

1) I would like the activity logging to happen regardless of whether it is a success or failure
2) There should be examples in  api_post_deliver_item_spec.rb verifying that ActivityLogger.api_scan_item and ActivityLogger.api_ship_item are called

**Comment by rfox2** [comment]
@jaronkk I think I've covered everything you wanted. Look OK?

File: app/services/api_post_deliver_item.rb (hunk 1 above)
Comment (rfox2) [inline app/services/api_post_deliver_item.rb]: Done

File: app/services/api_post_deliver_item.rb (hunk 2 above)
Comment (rfox2) [inline app/services/api_post_deliver_item.rb]: Done

File: app/services/api_post_deliver_item.rb [hunk 3]
```diff
@@ -13,14 +13,25 @@ def initialize(match:)
   def post_data!
     response = ApiHandler.post(action: delivery_type, params: params)
     if response.success?
+      log_activity(response)
       response
     else
+      log_activity(response)
```
Comment (jaronkk) [inline app/services/api_post_deliver_item.rb]: Can you move this to before the if statement? Then you won't need to have it in here twice.

Comment (rfox2) [inline app/services/api_post_deliver_item.rb]: Done

**Comment by jaronkk** [comment]
:+1:

**Pull request merged**